### PR TITLE
Move Galois Field from Nova

### PR DESCRIPTION
**Pull request by mpenciak**
The moved `GaloisField` file is very close to the file found in the Nova repository, but there were a few additions and deletions in the transfer to make working with Galois Fields more ergonomic.

### CODE REVIEW

File: YatimaStdLib/GaloisField.lean [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+instance [GaloisField K] : Div K where
+  div := divis
+
+instance [GaloisField K] : Sub K where
+  sub := minus
+
+/-- An `O(log n)` implementation of `galPow` -/
+def fastPow [Mul K] [OfNat K (nat_lit 1)] (x : K) (n : Nat) : K := 
+  let binExp := n.toBits
```
Comment (DanielRrr) [inline YatimaStdLib/GaloisField.lean:54]: Why don't we replace all those constraints with `[GaloisField K]`?

File: YatimaStdLib/GaloisField.lean [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    | 1 => acc
+    | n + 1 => getSquares ((acc.headD x) * (acc.headD x) :: acc) n
+
+def galPow [GaloisField K] : K → Nat → K
+  | _, 0 => 1
+  | x, (k + 1) => x * (galPow x k)
+
+-- TODO: Replace this with `fastPow`?
+instance [GaloisField K] : HPow K Nat K where
+  hPow := galPow
```
Comment (DanielRrr) [inline YatimaStdLib/GaloisField.lean:69]: `galPow` actually already utilises `galPow`, so the comment above doesn't seem needed

Comment (mpenciak) [reply YatimaStdLib/GaloisField.lean:69]: The question was whether to replace `galPow` with `fastPow` in the `HPow` instance for `K`... I think I've decided to keep it as is, and instead offer `fastPow` as an alternative when we need it (for example, computing the Frobenius in a field of size ~2^256, `fastPow` is almost 100% required)

File: YatimaStdLib/GaloisField.lean (hunk 1 above)
Comment (mpenciak) [reply YatimaStdLib/GaloisField.lean:54]: I actually need `fastPow` in the instance declaration of `GaloisField` for `Extension K P` (to define the Frobenius), so unfortunately the way I want to use it I don't get to use the `GaloisField K` instance yet :(

Comment (DanielRrr) [reply YatimaStdLib/GaloisField.lean:54]: OK, let's leave it as it is

**Pull request merged**